framework/Metrics/DTW.py

```diff
@@ -61,25 +78,15 @@ def _localReadMoreXML(self, xmlNode):
       @ In, xmlNode, xml.etree.Element, Xml element node
       @ Out, None
     """
-    self.requiredKeywords = set(['order','localDistance'])
-    self.wrongKeywords = set()
-    for child in xmlNode:
-      if child.tag == 'order':
-        if child.text in ['0','1']:
-          self.order = float(child.text)
-        else:
-          self.raiseAnError(IOError,'DTW metrics - specified order ' + str(child.text) + ' is not recognized (allowed values are 0 or 1)')
-        self.requiredKeywords.remove('order')
-      if child.tag == 'localDistance':
-        self.localDistance = child.text
-        self.requiredKeywords.remove('localDistance')
-      if child.tag not in self.requiredKeywords:
-        self.wrongKeywords.add(child.tag)
-
-    if self.requiredKeywords:
-      self.raiseAnError(IOError,'The DTW metrics is missing the following parameters: ' + str(self.requiredKeywords))
-    if not self.wrongKeywords:
-      self.raiseAnError(IOError,'The DTW metrics block contains parameters that are not recognized: ' + str(self.wrongKeywords))
+    paramInput = DTW.getInputSpecification()()
+    paramInput.parseNode(xmlNode)
+    for child in paramInput.subparts:
+      if child.getName() == "order":
+        self.order = child.value
+        if self.order not in [0,1]:
```

Comment:
you do not need this check.
You should use the enumerator type instead of the IntegerType => no check required since it is done directly in the InputData